File: backups/20260725-175826/api/history.py
import json
from pathlib import Path
from datetime import datetime


HISTORY_FILE = Path(
    "agents/data/history.json"
)


def load_history():
    if not HISTORY_FILE.exists():
        return []

    return json.loads(
        HISTORY_FILE.read_text()
    )


def save_execution(
    message,
    provider,
    response
):
    history = load_history()

    history.append({
        "time": datetime.utcnow().isoformat(),
        "message": message,
        "provider": provider,
        "response": response
    })

    HISTORY_FILE.write_text(
        json.dumps(
            history,
            indent=2
        )
    )

    return history[-1]


def get_history():
    return load_history()

File: apps/web/src/api/history.ts
const API_URL =
  import.meta.env.VITE_ROUTER_API_URL ||
  "http://127.0.0.1:8000"


export type HistoryItem = {
  time: string
  message: string
  provider: string
  response: string
}


export async function getHistory(): Promise<HistoryItem[]> {

  const response =
    await fetch(
      `${API_URL}/history`
    )

  return response.json()
}
